src/Test/Utilities/Portable/project.json

```diff
@@ -9,16 +9,14 @@
     "System.Diagnostics.Tools": "4.0.1",
     "System.IO.FileSystem": "4.0.1",
     "System.Reflection.TypeExtensions": "4.1.0",
+    "System.Text.RegularExpressions": "4.1.0",
```

Comment:
Interesting difference, here's what happened.

RegularExpressions is an implementation-only dependency of `System.Xml.ReaderWriter`.  As such, that package references it with `"exclude": "compile"`.

Previously you were pulling in the "dotnet" section of [xunit.assert](https://www.nuget.org/packages/xunit.assert/2.1.0)
```
Left:  'c:\Users\ericstj\Desktop\before%2520project.lock.json'
Right: 'c:\users\ericstj\Desktop\after%2520project.lock.json'
714  -       "xunit.assert/2.1.0": {
    <--        "dependencies": {
    <--          "System.Collections": "4.0.0",
    <--          "System.Diagnostics.Debug": "4.0.0",
    <--          "System.Globalization": "4.0.0",
    <--          "System.Linq": "4.0.0",
    <--          "System.ObjectModel": "4.0.0",
    <--          "System.Reflection": "4.0.0",
    <--          "System.Reflection.Extensions": "4.0.0",
    <--          "System.Runtime": "4.0.0",
    <--          "System.Runtime.Extensions": "4.0.0",
    <--          "System.Text.RegularExpressions": "4.0.0",
    <--          "System.Threading.Tasks": "4.0.0"
    <--        },
```
This referenced `System.Text.RegularExpressions` without any `"exclude": "compile"`.

As such, you were getting RegularExpressions at compile time as a result of the `xunit.assert` dependency.  When you removed the `dotnet` import you no longer got this dependency group from `xunit.assert` so the only remaining reference was the one from  `System.Xml.ReaderWriter`, thus the placeholder.